streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -119,6 +123,7 @@ public void put(final Bytes key,
         validateStoreOpen();
         lock.writeLock().lock();
         try {
+            validateStoreOpen();
```

Comment:
We need to check again after we grab the lock, otherwise the store might get closed after we check but before we grab the lock. Once we get the lock, we're guaranteed that this block is serialized wrt `close()`. But we can still check beforehand to avoid grabbing the lock if it is closed.

Reply:
Why would it not be sufficient to only check after we got the lock?

Reply:
It would be sufficient. I assumed that we were checking before the lock to avoid synchronization overhead, so I left that in place.